rules/windows/command_and_control_certutil_network_connection.toml

```diff
@@ -31,6 +31,28 @@ services.path FROM services JOIN authenticode ON services.path = authenticode.pa
 authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != 'trusted'
 """
 
+[[transform.investigate]]
+label = "Alerts associated with the user in the last 48h"
+providers = """
+[[{"field":"event.kind","excluded":false,"queryType":"phrase","value":"signal","valueType":"string"},{"field":"user.id","excluded":false,"queryType":"phrase","value":"{{user.id}}","valueType":"string"}]]
+"""
+relativeFrom = "now-48h/h"
+relativeTo = "now"
```

Comment:
What about breaking up the object like this:

```python
[transform.investigate]
label = "Alerts associated with the user in the last 48h"
relativeFrom = "now-48h/h"
relativeTo = "now"

[transform.investigate.providers]

[[transform.investigate.providers.field_info]]
field = "event.kind"
excluded = false
queryType = "phrase"
value = "signal"
valueType = "string"

[[transform.investigate.providers.field_info]]
field = "user.id"
excluded = false
queryType = "phrase"
value = "{{user.id}}"
valueType = "string"
```

Reply:
It looks better, but maybe it is more error-prone this way?

Reply:
Potentially, but at least we can validate each field by schema this way.

Reply:
Yeah, could we have a detection_rules utility to convert it from the raw syntax to this?

Reply:
we added a command to help convert osquery for this same reason

Reply:
I've adjusted the format in https://github.com/elastic/detection-rules/pull/3080/commits/0f09f053ff4bcda1b83197b60f99f8e418bb672d

Reply:
I think I would actually prefer the flattened way for 2 reasons
- toml dumps it that way
- keeps the rule file from getting so long that the actual `rule` is all the way at the bottom


example:

```
!investigate{"label":"Alerts associated with the host in the last 48h","providers":[[{"field":"event.kind","excluded":false,"queryType":"phrase","value":"signal","valueType":"string"},{"field":"host.name","excluded":false,"queryType":"phrase","value":"{{host.name}}","valueType":"string"}]],"relativeFrom":"now-48h/h","relativeTo":"now"}
```

dumps to this

```toml
label = "Alerts associated with the host in the last 48h"
providers = [ [ [ "field", "excluded", "queryType", "value", "valueType",], [ "field", "excluded", "queryType", "value", "valueType",],],]
relativeFrom = "now-48h/h"
relativeTo = "now"
```

it will still be fully validated per our defined schema

Reply:
Any reason why there is so many brackets? `[ [ [ ... ] ] ]` Is it generated like that or just a typo?